plugins/c8y_configuration_plugin/src/config.rs

```diff
@@ -1,90 +1,177 @@
 use c8y_smartrest::topic::C8yTopic;
 use mqtt_channel::Message;
 use serde::Deserialize;
+use std::borrow::Borrow;
 use std::collections::HashSet;
 use std::fs;
 use std::hash::{Hash, Hasher};
-use std::path::PathBuf;
+use std::path::Path;
+use tedge_utils::file::FilePermissions;
 use tracing::{info, warn};
 
-#[derive(Deserialize, Debug, Eq, PartialEq, Default)]
+#[derive(Deserialize, Debug, Default)]
 #[serde(deny_unknown_fields)]
+struct RawPluginConfig {
+    pub files: Vec<RawFileEntry>,
+}
+
+#[derive(Deserialize, Debug, Default, PartialEq)]
+#[serde(deny_unknown_fields)]
+pub struct RawFileEntry {
+    pub path: String,
+    #[serde(rename = "type")]
+    config_type: Option<String>,
+    user: Option<String>,
+    group: Option<String>,
+    mode: Option<u32>,
+}
+
+#[derive(Debug, Eq, PartialEq, Default, Clone)]
 pub struct PluginConfig {
     pub files: HashSet<FileEntry>,
 }
 
-#[derive(Deserialize, Debug, Eq, Default, Clone)]
-#[serde(deny_unknown_fields)]
+#[derive(Debug, Eq, Default, Clone)]
 pub struct FileEntry {
-    path: String,
+    pub path: String,
+    config_type: String,
+    pub file_permissions: FilePermissions,
 }
 
 impl Hash for FileEntry {
     fn hash<H: Hasher>(&self, state: &mut H) {
-        self.path.hash(state);
+        self.config_type.hash(state);
     }
 }
 
 impl PartialEq for FileEntry {
     fn eq(&self, other: &Self) -> bool {
-        self.path == other.path
+        self.config_type == other.config_type
     }
 }
 
-impl FileEntry {
-    pub(crate) fn new(path: String) -> Self {
-        Self { path }
+impl Borrow<String> for FileEntry {
+    fn borrow(&self) -> &String {
+        &self.config_type
     }
 }
 
-impl PluginConfig {
-    pub fn new(config_file_path: PathBuf) -> Self {
-        let config_file_path_str = config_file_path.as_path().display().to_string();
-        Self::read_config(config_file_path).add_file(config_file_path_str)
+impl FileEntry {
+    pub fn new(
+        path: String,
+        config_type: String,
+        user: Option<String>,
+        group: Option<String>,
+        mode: Option<u32>,
+    ) -> Self {
+        Self {
+            path,
+            config_type,
+            file_permissions: FilePermissions { user, group, mode },
+        }
     }
+}
 
-    fn read_config(path: PathBuf) -> Self {
-        let path_str = path.as_path().display().to_string();
+impl RawPluginConfig {
+    fn new(config_file_path: &Path) -> Self {
+        let path_str = config_file_path.display().to_string();
         info!("Reading the config file from {}", path_str);
-        match fs::read_to_string(path) {
+        match fs::read_to_string(config_file_path) {
             Ok(contents) => match toml::from_str(contents.as_str()) {
                 Ok(config) => config,
-                _ => {
-                    warn!("The config file {} is malformed.", path_str);
+                Err(err) => {
+                    warn!("The config file {path_str} is malformed. {err}");
```

Comment:
This is more an `error!` than a `warn!`.

Reply:
Same comment, for all the `warn!` usages in this file. Something is wrong and must be fixed, even if the program doesn't stop.